src/DDTrace/Integrations/Redis/RedisIntegration.php

```diff
@@ -0,0 +1,835 @@
+<?php
+
+namespace DDTrace\Integrations\Redis;
+
+use DDTrace\Integrations\Integration;
+use DDTrace\Tag;
+use DDTrace\Type;
+use DDTrace\GlobalTracer;
+use DDTrace\Util\TryCatchFinally;
+
+const VALUE_PLACEHOLDER = "?";
+const VALUE_MAX_LEN = 100;
+const VALUE_TOO_LONG_MARK = "...";
+const CMD_MAX_LEN = 1000;
+
+class RedisIntegration extends Integration
+{
+    const NAME = 'redis';
+
+    /**
+     * @var array
+     */
+    private static $connections = [];
+
+    /**
+     * @var self
+     */
+    private static $instance;
+
+    /**
+     * @return self
+     */
+    public static function getInstance()
+    {
+        if (null === self::$instance) {
+            self::$instance = new self();
+        }
+        return self::$instance;
+    }
+
+    /**
+     * @return string The integration name.
+     */
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    /**
+     * Static method to add instrumentation to the Redis library
+     */
+    public static function load()
+    {
+        if (!extension_loaded('redis')) {
+            // `curl` extension is not loaded, if it does not exists we can return this integration as
+            // not available.
+            return Integration::NOT_AVAILABLE;
+        }
+
+        dd_trace('Redis', 'connect', function () {
+            $args = func_get_args();
+            $scope = GlobalTracer::get()->startIntegrationScopeAndSpan(
+                RedisIntegration::getInstance(),
+                'Redis.connect'
+            );
+            $span = $scope->getSpan();
+            $span->setTag(Tag::SPAN_TYPE, Type::CACHE);
+            $span->setTag(Tag::SERVICE_NAME, 'redis');
+            $span->setTag(Tag::RESOURCE_NAME, 'Redis.connect');
+
+
+            $thrown = null;
+            try {
+                $result = call_user_func_array([$this, 'connect'], $args);
+            } catch (\Exception $e) {
+                RedisIntegration::setErrorOnException($span, $e);
+                $thrown = $e;
+            }
+            $scope->close();
+
+            if ($thrown) {
+                throw $thrown;
+            }
+
+            return $result;
+        });
+
+        dd_trace('Redis', 'getSet', function () {
```

Comment:
We typically don't trace all the possible methods from an extension. Instead we try to only trace methods that perform I/O operations or are otherwise "slow" relative to the other methods. Would you be able to whittle this list of methods down to just the ones that are typically slower for this extension? :)